File: src/main/kotlin/a_basic_control_structures/exercise_10.kt
package a_basic_control_structures

// Write an algorithm that reads the width and height of a wall, calculates and displays the area to be painted and the amount of paint needed for the job, considering that each liter of paint paints an area of 2 square meters.

fun main() {
    print("Wall width: ")
    val width: Double = readln().toDouble()

    print("Wall height: ")
    val height: Double = readln().toDouble()

    val measure = width * height
    val _measure = String.format("%.0f", measure)

    val volume: Double = measure * 2
    val _volume = String.format("%.0f", volume)

    println("You have $_measure square meters to paint. \nIt will be necessary $_volume liters of paint.")
}

File: Dart/a_basic_control_structure/exercise_10.dart
// Write an algorithm that reads the width and height of a wall, calculates and displays the area to be painted and the amount
// of paint needed for the job, considering that each liter of paint paints an area of 2 square meters.

import 'dart:io';

void main() {
  double width = getInput('wall width: ');
  double height = getInput('wall height: ');

  calculateAreaAndPaint(width, height);
}

double getInput(String prompt) {
  while (true) {
    stdout.write(prompt);
    String? input = stdin.readLineSync();
    if (input == null) {
      print('Invalid input. Please enter a number.');
      continue;
    }
    try {
      return double.parse(input);
    } on FormatException {
      print('Invalid input. Please enter a number.');
    }
  }
}

void calculateAreaAndPaint(double width, double height) {
  double measure = width * height;
  double volume = measure * 2;

  print(
      'You have $measure square meters to paint. \nIt will be necessary $volume liters of paint.');
}
